### PR TITLE
Fix error checking when reading empty files with parallel input

### PR DESCRIPTION
**Pull request by ericfischer**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -367,7 +367,7 @@ void do_read_parallel(char *map, long long len, long long initial_offset, const
 		void *retval;
 
 		if (pthread_join(pthreads[i], &retval) != 0) {
-			perror("pthread_join");
+			perror("pthread_join 370");
```
Comment (Hi-Angel) [inline main.cpp:370]: `pthread_join` doesn't modify `errno`, instead it returns an error that has to be passed to `strerror()` *(alternatively, `man pthread_create` has example where it assigns the return value to `errno`, and then calls `perror()`)*.

Comment (Hi-Angel) [reply main.cpp:370]: *(I noticed the code is still there in `main.cpp`, so just decided to note that. I don't really know what project this is, was just looking on the internet for examples why `pthread_join` could return `INV_ARG` whereas the 1st arg is a valid tid, and the second is just `NULL`, so basically both are valid)*.

Comment (e-n-f) [reply main.cpp:370]: Thanks, I'll take a look.

Comment (Hi-Angel) [reply main.cpp:370]: Just found more explicit documentation, decided to share for posterity: `man 7 pthreads` claims that none of the pthreads functions set `errno`. Quoting:

> Note that the pthreads functions do not set errno.

**Pull request merged**